---
layout: default
title: Website of Niels W. Hanson
---

# Curriculum Vitae

This document is available as a [downloadable pdf](docs/hanson_cv.pdf) or as a more [summarized resumé](docs/hanson_resume.pdf).

## Technical Skills

* **Basic** - MPI, OpenCL, CUDA
* **Intermediate** - HTML, CSS, JavaScript, PHP, Perl, d3.js, Git, MATLAB, LaTeX, Hadoop, GLPK
* **Advanced** - Java, C++, Python, R

## Education

* 2011-2015 (Expected) - Ph.D. Bioinformatics - University of British Columbia
* 2006-2011 - B.Sc. Computer Science (Statistics Minor) - University of British Columbia

## Awards
* 2012 - Four Year Doctoral Fellowship (4YF) - University of British Columbia
    
    Awarded to incoming doctoral graduate students awarded on academic excellence, upon the recommendation of the graduate program.

## Experience

* 2011--Now - Bioinformatics Ph.D. Candidate - Steven J. Hallam Laboratory
	
	Primary research focuses on the development of MetaPathways, a pipeline developed in Python and C++ for the Big Data processing and analysis challenges of next-generation environmental sequence information. The software accommodates a wide range of experimental conditions, integrates with HPC resources via a master-worker model, and is controlled locally via user-friendly GUI. Moreover, I am involved in all areas of lab data analysis and research, and have utilized a combination of Python and R to develop bespoke solutions to a variety of analytical challenges.

* 2011 - Software Developer - UBC Microbiology and Immunology

	Developer of RameyDB, a database-driven web-application for the cataloging of bacterial strains and plasmids at the University of British Columbia. Utilized the LAMP stack, JavaScript, and PHP to create an easy-to-use web interface. Implemented a number of features to improve usability including an advanced search form, user search histories, and intelligent autocomplete to ease data entry.

* 2010 - Bioinformatics Intern - Evan E. Eichler Laboratory, University of Washington
	
	Summer research position involved the development of a database-driven hu- man genome visualization software for analysis of autistic genomes. Designed and implemented MySQL back-end database and a Java Swing-based GUI and visualization environment for genetic variants predicted via a Hidden Markov Model. Designing efficient layout algorithms allowed for the rapid analysis and validation of important genomic alterations, replacing a previously tedious and disconnected analysis pipeline.

* 2009 - Software Developer - Centre for Microbial Disease and Immunity Research

	Developed a data-base driven web-application PaIntDB, a biological database for the interactive query and exploration of protein-protein interaction data using Jakarta Struts and JavaScript with a LAMP stack back end. Performed multiple rounds of development and consultation with microbiologist and UBC and SFU stakeholders.

* 2008 - Software Developer - UBC Information Technology

	Key developer of a web-based events calendar UBCEvents. Designed, styled, and implemented the Web UI through CSS, HTML, and JavaScript. Developed and modified the underlying open source Bedework XSLT code to parse and translate XML objects retrieved from an Oracle database into structured web- pages. Designed statistical scripts using Perl and R to parse logs and summarize calendar use.

## Publications

For an up-to-date list see my [google scholar page](goo.gl/aYo3Wa).

* **Niels W. Hanson**, Kishori M. Konwar, Shang-Ju Wu, Steven J. Hallam. *Introduction to the Analysis of Environmental Sequence Information Using MetaPathways*, Computational Methods for Next Generation Sequencing Data Analysis. Book Chapter. Wiley Series in Bioinformatics. In Press.

* Dongjae Kim, Kishori M. Konwar, **Niels W. Hanson**, Steven J. Hallam. *Koonkie: An Automated Software Tool for Processing Environmental Sequence Information using Hadoop*, ASE BigData 2014. Harvard University, December 14-16, 2014.[http://www.ase360.org/handle/123456789/164](http://www.ase360.org/handle/123456789/164)

* **Niels W. Hanson**, Kishori M. Konwar,  Alyse K. Hawley, Tomer Altman, Peter D. Karp, Steven J. Hallam. *Metabolic pathways for the whole community*, BMC Genomics. July 2014. [doi:10.1186/1471-2164-15-619](http://dx.doi.org/10.1186/1471-2164-15-619)
		  
* **Niels W. Hanson**, Kishori M. Konwar, Shang-Ju Wu, Steven J. Hallam. *MetaPathways v2.0: A master-worker model for environmental Pathway/Genome Database construction on grids and clouds.* Proceedings of the 2014 IEEE Conference on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB 2014), Honolulu, HI, USA, May 21-24, 2014. [doi:10.1109/CIBCB.2014.6845516](http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=6845516)

* Jody J. Wright, Keith Mewis, **Niels W. Hanson**, Kishori M. Konwar, Kendra R. Maas, Steven J. Hallam. *Genomic properties of Marine Group A bacteria indicate a role in the marine sulfur cycle*, ISME Journal. September 2013. [doi:10.1038/ismej.2013.152](http://dx.doi.org/10.1038/ismej.2013.152)

* Dongshan An, Sean Michael Caffrey, Jung Soh, Akhil Agrawal, Damon Brown, Karen Budwill, Xiaoli Dong, Peter F. Dunfield, Julia Foght, Lisa M. Gieg, Steven J. Hallam, **Niels W. Hanson**, Zhiguo He, Thomas R. Jack, Jonathan Klassen, Kishori M. Konwar, Eugene Kuatsjah, Carmen Li, Steve Larter, Verlyn Leopatra, Camilla L Nesbø, Thomas B.P. Oldenburg, Antoiné P. Page, Esther Ramos-Padron, Fauziah Rochman, Ali Saidi-Mehrabad, Christoph W. Sensen, Payal Sipahimalani, Young C. Song, Sandra Wilson, Gregor Wolbring, Ginny Wong, Gerritt Voordouw.  Metagenomics of Hydrocarbon Resource Environments Indicates Aerobic Taxa and Genes to be Unexpectedly Common. *Environmental Science & Technology*, September 2013 [doi:10.1021/es4020184](http://dx.doi.org/10.1021/es4020184)

* Brandon K. Swan, Ben Tupper, Alexander Sczyrba, Federico M. Lauro, Manuel Martinez-Garcia, José M. González, Haiwei Luo, Jody J. Wright, Zachary C. Landry, **Niels W. Hanson**, Brian P. Thompson, Nicole J. Poulton, Patrick Schwientek, Silvia G. Acinas, Stephen J Giovannoni, Mary Ann Moran, Steven J. Hallam, Ricardo Cavicchioli, Tanja Woyke, Ramunas Stepanauskas. *Prevalent genome streamlining and latitudinal divergence of planktonic bacteria in the surface ocean*, Proceedings of the National Academy of Sciences. July 2013. [doi:10.1073/pnas.1304246110](http://dx.doi.org/10.1073/pnas.1304246110)

* Kishori M. Konwar, **Niels W. Hanson**, Antoine P. Pagé, Steven J. Hallam. *MetaPathways: a modular pipeline for constructing pathway/genome databases from environmental sequence information*, BMC Bioinformatics. June 2013.[doi:10.1186/1471-2105-14-202](http://dx.doi.org/10.1186/1471-2105-14-202)

* W. Evan Durno, **Niels W. Hanson**, Kishori M. Konwar, Steven J. Hallam. *Expanding the boundaries of local similarity analysis*, BMC Genomics. February 2013. [doi:10.1186/1471-2164-14-S1-S3](http://dx.doi.org/10.1186/1471-2164-14-S1-S3)

* Kaston Leung, Hans Zahn, Timothy Leaver, Kishori M. Konwar, **Niels W. Hanson**, Antoine P. Pagé, Chien-Chi Lo, Patrick S. Chain, Steven J. Hallam, Carl L. Hansen. *A programmable droplet-based microfluidic device applied to multiparameter analysis of single microbes and microbial communities*, Proceedings of the National Academy of Sciences. May 2012. [doi:10.1073/pnas.1106752109](http://dx.doi.org/10.1073/pnas.1106752109)

## In-review

* Aria S. Hahn, **Niels W. Hanson**, Dongjae Kim, Kishori M. Konwar, Steven J. Hallam. *Assembly independent functional annotation of short-read data using SOFA: Short-ORF Functional Annotation*, 11th International Symposium on Bioinformatics Research and Applications (ISBRA 2015). In review.

* Kishori M. Konwar, **Niels W. Hanson**, Maya P. Bhatia, Dongjae Kim, Shang-Ju Wu, Aria S. Hahn, Connor Morgan-Lang, Hiu Kan Cheung, Steven J. Hallam. *MetaPathways v2.5: Quantitative functional, taxonomic, and usability improvements*, Oxford Bioinformatics. In review.

## Conferences & Talks

* **Niels W. Hanson**, Kishori M. Konwar, Shang-Ju Wu, Steven J. Hallam. *MetaPathways v2.0: A master-worker model for environmental Pathway/Genome Database construction on grids and clouds.* IEEE Conference on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB 2014). Presentation. Honolulu, HI, USA.   May 21–24 2013.

* Frances K. Russell, **Niels W. Hanson**, Kishori M. Konwar, Steven J. Hallam. *Hierarchical and High-Performance clustering and annotation for large protein sequence databases.* Poster. Ottawa, ON, Canada. June 2–6 2013.

* **Niels W. Hanson**, W. Evan Durno, Kishori M. Konwar, Steven J. Hallam. *IMPROV: An integrated MetaPRO-teomics viewer.* Vancouver, BC, Canada. International Symposium on Microbial Ecology (ISME14). Poster. Copenhagen, Denmark. August 19–24 2013.

* **Niels W. Hanson**, Antoine P. Pagé, Kishori M. Konwar, Charles G. Howes, Steven J. Hallam. *Metabolic Interaction Networks for the Whole Community*. Poster. Melbourne, Australia. January 17-19 2012.

## Journal & Conference Reviewing

* 2015 IEEE Conference on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB 2015)
* 11th International Symposium on Bioinformatics Research and Applications (ISBRA 2015)
* 2014 IEEE Conference on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB 2014)

## Teaching

* Summer 2014 - Teaching Assistant - Marine Biological Laboratory (MBL)
	
	Teaching Assistant for the prestegious Strategies and Techniques for Analyzing Microbial Population Structure (STAMPS) course offered at the Marine Biological Laboratory (MBL) in Woods Hole, Massachusetts. Lectured and conducted tutorials to more than 60 post-docs, graduate students, and professors on the use of MetaPathways to process and annotate environmental sequence information, and downstream statistical analysis and visualization using the R, ggplot2, dplyr, and Pathway Tools software.

* Spring 2014 - Teaching Assistant - University of British Columbia
	
	Graduate teaching assistant of the Problem Based Learning in Bioinformatics (BIOF 520) graduate class. Responsibilities included assisting with lectures, designing and facilitating the Human Microbiome PBL learning module, grading assignments, and general class administration.

* 2013-2015 - Teaching Assistant - University of British Columbia 
	
	Conducted 1–2 weeks of lectures annually in the Microbial Ecological Genomics class (MICB 425). Topics included the functional and taxonomic analysis of en- vironmental sequence information, with a focus on using the MetaPathways pipeline. Analytical topics included down-stream statistical analysis including Hierarchical Clustering, Principal Component Analysis (PCA), Non-metric Multi-dimensional Scaling (NMDS), and general visualization in the R statistical envi- ronment using ggplot2 and reproducible analysis using RMarkdown.

## Projects

* **De-Confusion Tables**. An interactive R shiny app for exploring performance statistics of a binary classifier. Final project. Developing Data Products. John Hopkins University Coursera Class. July 2014. [http://goo.gl/Fl78Qh](http://goo.gl/Fl78Qh)
* **D3.js visualizations**:
    * *World Map*: World map displaying metagenomic sequencing data. http://goo.gl/WVjdAv
    * *Sunburst Plot*: Radial tree-map showing hierarchal classification of global metagenomes. http://goo.gl/3LlXIq
    * *Bubble Tree*: Modified dendrogram to show taxonomy across multiple samples. http://goo.gl/2SBVsp
    * *Bubble Plot*: A two-variable sortable bubble plot. http://goo.gl/Jfslu1
    * *Heatmap*: Heatmap: Two-variable heatmap with calculated marginal distributions. http://goo.gl/3dIurm
* **FastLCA R-Package**. An implementation the FastLCA method for local correlations in R. Utilizes the optimized C code and accepts Data Frames from R. Parallelized using Open-MP. [http://goo.gl/cPb1uP](http://goo.gl/cPb1uP).

## Interests

**Professional:** Data Analysis, Information Visualization, Machine Learning, Statistics, High-performance Computing, Web-app Creation, Software Design 

**Personal:** Running, Tennis, Badminton, Piano, Photography, Cooking

## References

Provided upon request.


