<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="shortcut icon" href="/img/favicon.ico" type="image/x-icon">
    <link rel="icon" href="/img/favicon.ico" type="image/x-icon">
    <meta property="og:image" content="http://www.nielshanson.com/img/nh_logo.png" />
    <link rel="image_src" href="http://www.nielshanson.com/img/nh_logo.png" />

    <title>Niels Hanson</title>

    <!-- Bootstrap core CSS -->
    <link href="/css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom styles for this template -->
    <link href="/css/jumbotron-narrow.css" rel="stylesheet">

    <!-- MyStyles -->
    <link href="/css/styles.css" rel="stylesheet">
	<!-- Mathjax -->
	<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>


    <!-- Just for debugging purposes. Don't actually copy these 2 lines! -->
    <!--[if lt IE 9]><script src="../../assets/js/ie8-responsive-file-warning.js"></script><![endif]-->
    <!--<script src="../../assets/js/ie-emulation-modes-warning.js"></script> -->

    <!-- IE10 viewport hack for Surface/desktop Windows 8 bug -->
    <!--<script src="../../assets/js/ie10-viewport-bug-workaround.js"></script> -->

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
    <!-- Google Analytics code-->
    <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','http://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-69343009-1', 'auto');
    ga('send', 'pageview');

    </script>
  </head>

  <body>
    <div class="container-fluid">
      <div class="header">
        <ul class="nav nav-pills pull-right">
		  <li><a href="/index.html">Home</a></li>
          <li><a href="/hanson_cv.html">CV</a></li>
		  <li><a href="/portfolio.html">Portfolio</a></li>
          <li><a href="/contact.html">Contact</a></li>
        </ul>
		<div class="iconheader">
			<a href="https://twitter.com/nielshanson"><img src="/img/twitter.png" alt="github" class="img-rounded" width=40></a>
			<a href="https://www.linkedin.com/profile/view?id=59493970"><img src="/img/linkedin.png" alt="linkedin" width=40 class="img-rounded"></a>
			<a href="https://github.com/nielshanson/"><img src="/img/github.png" alt="github" class="img-rounded" width=40></a>

		</div>

      </div>
  	<div class="jumbotron">
      <div class="row vertical-align">
        <div class="col-md-12 jumbotext">
        	<h1>Niels Hanson</h1>
        	<p class="lead">Data scientist and software developer.</p>
        </div>
      </div>
      <div class="row vertical-align">
        <div class="col-md-12 jumbotext">
        <img src="/img/nh_logo_trans.png" width="150px" class="centerandmiddle"></img>
        </div>
      </div>
  	</div>

<h1 id="portfolio">Portfolio</h1>

<p>This page highlights a selection of my software, analyses, and visualization projects.</p>

<h2 id="software">Software</h2>

<p>Here I&#39;ve highlighted some of the software projects I have been involved with.</p>

<div class="row">
<div class="col-sm-4">
<h3>MetaPathways</h3>
<img src="img/mp_logo.png", alt="MetaPathways pipeline" style="width: 140px; height: 140px;">
<p>A high-performance pipeline for the systematic processing and analysis of next-generation environmental datasets, allowing the systematic preprocessing, gene prediction, annotation, and comparison of thousands of metagenomic samples. Published in <a href="doi:10.1186/1471-2105-14-202">BMC Bioinformatics</a>.</p>
<p><a class="btn btn-default" href="https://github.com/hallamlab/metapathways" role="button">View details »</a></p>
</div>

<div class="col-sm-4">
<h3>Master-worker</h3>
<img src="img/mw_logo.png", alt="MetaPathways pipeline" style="width: 140px; height: 140px;">
<p>Designed a distributed Master-worker algorithm for the scheduling of compute tasks to multilple HPC computational grids. Implemented as a feature of the <a href="https://github.com/hallamlab/metapathways2">MetaPathways v2.0</a> to process computationally intensive homology-search tasks. Published in <a href="http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=6845516">IEEE CIBCB 2014</a>.</p>
<p><a class="btn btn-default" href="https://github.com/hallamlab/metapathways2" role="button">View details »</a></p>
</div>

<div class="col-sm-4">
<h3>FastLSA</h3>
<a href="http://hallam.microbiology.ubc.ca/fastLSA/"><img src="img/lsa_logo.png", alt="MetaPathways pipeline" style="width: 140px; height: 140px;"></a>
<p>FastLSA is a correlation detection method specializing the detection of partial, leading, or lagging correlation, particilar within time-series. P-values are calculated using a <a href="http://www.biomedcentral.com/1471-2164/14/S1/S3">closed-form approximation</a>, implemented in C, and paralleized with p-threads; hundreds of times faster than previous implementations. Published in <a href="http://dx.doi.org/10.1186/1471-2164-14-S1-S3">BMC Genomics</a>.</p>
<p><a class="btn btn-default" href="http://hallam.microbiology.ubc.ca/fastLSA/" role="button">View details »</a></p>
</div>

<p></div></p>

<h2 id="analytics">Analytics</h2>

<p>I&#39;ve been involved in a number of analysis of unstructured datasets from beginning to end. Applying a number of statistical and machine learning models and the <a href="http://ggplot2.org/">ggplot2</a> visualization framework. I am an avid user of  <a href="http://yihui.name/knitr/">Knitr</a>, <a href="http://rmarkdown.rstudio.com/">RMarkdown</a>, and <a href="http://www.rstudio.com/">RStudio</a> for reproducible data analysis. </p>

<div class="row">
<div class="col-sm-4">
<h3>Single Cell Genomes</h3>
<a href="https://github.com/nielshanson/microfluidic_single_cells"><img src="img/single_logo.png" class="img-rounded" alt="Single cells" style="width: 140px; height: 140px;"></a>
<p>Analysis combined three taxonomic identification methods, <a href="http://ab.inf.uni-tuebingen.de/software/megan/">MEGAN</a>, <a href="http://mltreemap.org/">ML-TreeMap</a>, and <a href="http://eggnog.embl.de/version_4.0.beta/">EggNOG</a>, to compare single-cell samples from three different environments, separated <a href="http://www.pnas.org/content/109/20/7665">using microfluidic device</a>. Techniques used included Gaussing <a href="http://en.wikipedia.org/wiki/Kernel_density_estimation">Kernel Density Estimation</a> and <a href="http://en.wikipedia.org/wiki/Hierarchical_clustering">Hierarchical Clustering</a>. Results visualized in R using the <a href="http://cran.r-project.org/web/packages/lattice/index.html">lattice</a> package. Results published in <a href="http://www.pnas.org/content/109/20/7665">PNAS</a>.</p>
<p><a class="btn btn-default" href="https://github.com/nielshanson/microfluidic_single_cells" role="button">View details »</a></p>
</div>

<div class="col-sm-4">
<h3>Metabolic Pathways</h3>
<img class="img-circle" src="img/pathways_logo.png" class="img-rounded" alt="Metabolic pathways" style="width: 140px; height: 140px;">
<p>Utilized MetaPathways to re-evaluate <a href="http://hahana.soest.hawaii.edu/hot/">Hawaii Ocean-time series</a> samples, providing guidelines for the analysis of predicted metabolic pathways from environmental samples. R and ggplot2 were used to compare activity of metabolic pathways biogeochemical variables like ocean depth and salinity. Developed a novel weighted distance to cacluate taxonomic variance within pathways. Published in <a href="http://dx.doi.org/10.1186/1471-2164-15-619">BMC Genomics</a>.</p>
<p><a class="btn btn-default" href="#" role="button">View details »</a></p>
</div>

<div class="col-sm-4">
<h3>Read Mapping</h3>
<img src="img/rpkm_logo.png" class="img-rounded" alt="Read Mapping" style="width: 140px; height: 140px;">
<p>An improvement in <a href="https://github.com/hallamlab/metapathways2">MetaPathways v2.5</a> is the ability to map reads to assembled sequences to estimate abundance. In this analysis we evaluate the variance of read-mapping against simple gene counting. Fitting a linear model mapped vs gene counts in metagenomic samples from the <a href="http://www.pac.dfo-mpo.gc.ca/science/oceans/data-donnees/line-p/index-eng.html">Pacific Ocean Line-P transact</a> showed variance is being corrected for in the mapped case. Published in <a href="http://dx.doi.org/10.1093/bioinformatics/btv361">Oxford Bioinformatics</a></p>
<p><a class="btn btn-default" href="https://github.com/hallamlab/mp_2_5_supplement" role="button">View details »</a></p>
</div>

<p></div></p>

<h2 id="visualization">Visualization</h2>

<p>I also develop interactive visualizations using the <a href="http://d3js.org/">d3.js</a> JavaScript library. <a href="http://bl.ocks.org/nielshanson">bl.ocks.org/nielshanson</a></p>

<div class="row">
    <div class="col-sm-4">
        <h3>World Map</h3>
        <a href="http://goo.gl/WVjdAv"><img src="img/world_map_logo.png" class="img-circle" alt="World Map" style="width: 140px; height: 140px;"></a>
        <p>World map of global metagenomes scaled by their sequencing abundance.</p>
        <p><a class="btn btn-default" href="http://goo.gl/WVjdAv" role="button">View details »</a></p>
    </div>
    <div class="col-sm-4">
        <h3>Sunburst Plot</h3>
        <a href="http://goo.gl/3LlXIq"><img src="img/sunburst_logo.png" class="img-circle" alt="Sunburst Plot" style="width: 140px; height: 140px;"></a>
        <p>Radial tree-map or Sunburst plot of global metagenomes classified by sampling category.</p>
        <p><a class="btn btn-default" href="http://goo.gl/3LlXIq" role="button">View details »</a></p>
    </div>
    <div class="col-sm-4">
        <h3>Bubble Tree</h3>
        <a href="http://goo.gl/2SBVsp"><img src="img/bubble_tree_logo.png" class="img-circle" alt="Bubble Tree" style="width: 140px; height: 140px;"></a>
        <p>Interative dendrogram to show taxonomy across multiple samples.</p>
        <p><a class="btn btn-default" href="http://goo.gl/2SBVsp" role="button">View details »</a></p>
    </div>
</div>

<div class="row">
    <div class="col-sm-4">
        <h3>Heatmap</h3>
        <a href="http://goo.gl/3dIurm"><img src="img/heatmap_logo.png" class="img-circle" alt="Heatmap" style="width: 140px; height: 140px;"></a>
        <p>Two-variable heatmap with calculated marginal distributions.</p>
        <p><a class="btn btn-default" href="http://goo.gl/3dIurm" role="button">View details »</a></p>
    </div>
    <div class="col-sm-4">
        <h3>Bubble Plot</h3>
        <a href="http://goo.gl/Jfslu1"><img src="img/bubble_plot_logo.png" class="img-circle" alt="Bubble Plot" style="width: 140px; height: 140px;"></a>
        <p>A two-variable sortable bubble plot with one-side marginal.</p>
        <p><a class="btn btn-default" href="http://goo.gl/Jfslu1" role="button">View details »</a></p>
    </div>
<div class="col-sm-4">
    <h3>Basic Barplot</h3>
        <a href="http://bl.ocks.org/nielshanson/a436b784293160dd4a20"><img src="img/basic_bar_logo.png", alt="Basic Barplot" class="img-circle" style="width: 140px; height: 140px;"></a>
        <p>Basic d3.js barplot to get started.</p>
        <p><a class="btn btn-default" href="http://bl.ocks.org/nielshanson/a436b784293160dd4a20" role="button">View details »</a></p>
    </div>
</div>

 <div class="footer">
    <div class="pull-left">&copy; Niels W. Hanson 2015</div><div class="pull-right">Updated October 2015</div>
  </div>

</div> <!-- /container -->

